app/scripts/list/list-rename-modal.controller.js

```diff
@@ -0,0 +1,33 @@
+(function () {
+	'use strict';
+
+	angular
+		.module('mustDoApp')
+		.controller('ListRenameModalController', ListRenameModalController);
+
+	ListRenameModalController.$inject = [
+		'$scope',
+		'$modalInstance',
+		'toaster',
+		'List',
+		'list'];
+
+	function ListRenameModalController($scope, $modalInstance, toaster, List, list) {
+		$scope.list = list;
+		$scope.name = list.name;
+		$scope.rename = rename;
+
+
+		function rename() {
+			var list = new List();
+			angular.extend(list, $scope.list);
+			list.name = list.name;
```

Comment:
It's not working properly - you have list with name "To Do" and you want to rename to "Thing to do". in Html you set new name to variable $scope.name but you are not using it here. So when i checked in Backend I received old name instead of new name :P 
I think it should be list.name = $scope.name;